src/wire/ipv4.rs

```diff
@@ -125,8 +125,11 @@ impl Cidr {
     ///
     /// # Panics
     /// This function panics if the prefix length is larger than 32.
-    pub fn new(address: Address, prefix_len: u8) -> Cidr {
-        assert!(prefix_len <= 32);
+    #[allow(clippy::no_effect)]
+    pub const fn new(address: Address, prefix_len: u8) -> Cidr {
+        // Replace with const panic (or assert) when stabilized
+        // see: https://github.com/rust-lang/rust/issues/51999
+        ["Prefix length should be <= 32"][(prefix_len > 32) as usize];
```

Comment:
This is a bit sketchy--is this a generally accepted workaround?

Reply:
From what I've read, yes. There's a reference to this technique in [this comment](https://github.com/rust-lang/rust/issues/51999#issuecomment-687624116) on the issue ticket referenced (in relation to the RustCrypto crate).

It's also the method used by 3rd party crates [static_assertions](https://github.com/nvzqz/static-assertions-rs/blob/master/src/bool.rs#L57) and [const_fn_assert](https://github.com/powlpy/const_fn_assert/blob/master/src/lib.rs#L86). I considered bringing one of them in- static_assertions seems to be more popular- but for such a simple macro it didn't seem worth the overhead. Those crates use the simplified `[()][bool_expr as usize]` form, but I've seen people use `["debug message"][bool_expr as usize]` to provide better feedback if the assert is triggered.